crates/emitter/src/scope_notes.rs

```diff
@@ -62,6 +62,27 @@ impl ScopeNoteList {
         ScopeNoteIndex::new(note_index)
     }
 
+    fn get_scope_hole_gcthing_index(&self, scope_note_index: &ScopeNoteIndex) -> GCThingIndex {
+        self.notes
+            .get(scope_note_index.index)
+            .expect("Scope note should exist")
+            .index
+    }
+
+    pub fn enter_scope_hole(
+        &mut self,
+        maybe_enclosing_scope_note_index: &Option<ScopeNoteIndex>,
+        body_index: GCThingIndex,
+        offset: BytecodeOffset,
+    ) -> ScopeNoteIndex {
+        let parent_index = ScopeNoteIndex::new(self.notes.len() - 1);
```

Comment:
this points the last scope we've entered before entering hole,
that can be sibling, not parent.

the parent should also be retrieved from the scope stack